DBMS/Assignment3/electronic.sql

```diff
@@ -0,0 +1,24 @@
+-- Query for to display all the categories along with its Parent category.
+-- Result is  sorted on Parent Category.
+-- Implementing self join and left outer join
+select 
+    node.name as node_name,
+    IFNULL(parent1.name, 'Top Category') as parent_name,
+    parent2.name as parent2_name,
+    parent3.name as parent3_name
+from
+    categories as node
+        left outer join
+    categories as parent1 ON parent1.id = node.parentid
+        left outer join
+    categories as parent2 ON parent2.id = parent1.parentid
+        left outer join
+    categories as parent3 ON parent3.id = parent2.parentid;
```

Comment:
ACTION: You have missed a query "Result should be sorted on Parent Category". Rewrite the above query and apply sorting on parent category.